### ISSUE TITLE
Data restore fails

### ISSUE THREAD

**Issue by simonsen23** [jensstein/oandbackup#211]
When I try to restore an app with oandbackup I get the following error:

"Appname": selinux_android_restorecon: not implemented for libselinux_vendor
"Appname": restorecon: restorecon failed: /data/user/0/"Appfolder"

The app gets restored but without any data. 

Phone: Oneplus 2 with LOS 15.1

**Comment by jensstein** [comment]
does this work for you?
https://drive.google.com/file/d/1hEagkOpnctLOAuWB3CPmTGul4Fxm3E_7/view?usp=sharing

**Comment by lee5783** [comment]
hi @jensstein i used your link and got this error at opening app 'couldn't find oab-utils - running in legacy mode, please report this to the developer'

**Comment by jensstein** [comment]
@lee5783 which architecture is your device? armv7?
please pick an appropriate apk here and try again:
https://gitlab.com/jensstein/oandbackup/builds/artifacts/master/browse?job=sign
if you still get the message, could you please open a new issue with information about your device and an attached logcat. also try executing ```ls -l /data/data/dk.jens.backup/files``` as root on your device.
thanks :)

**Comment by simonsen23** [comment]
@jensstein 
I get the same error as lee5783. 
Result from ls -l:
-rwxrw---- (...)
oab-utils

[log_oabutils.txt](https://github.com/jensstein/oandbackup/files/2127089/log_oabutils.txt)

PS: I'll open a new issue too.

**Comment by jensstein** [comment]
is there anything in oandbackup.log?

**Comment by simonsen23** [comment]
Nothing new as far as I know:

2018/06/24 - 13:24:19: selinux_android_restorecon: not implemented for
libselinux_vendor [ARD]
2018/06/24 - 13:24:19: restorecon: restorecon failed:
/data/user/0/de.swr.avp.ard [ARD]


jensstein <notifications@github.com> schrieb am So., 24. Juni 2018, 19:32:

**Comment by jensstein** [comment]
i just updated to lineageos 15.1 and now restorecon gives me the same error as you.
does this help:
https://gitlab.com/jensstein/oandbackup/-/jobs/77041353/artifacts/file/apks/arm64/oandbackup-debug.apk

**Comment by simonsen23** [comment]
I get a notification "Restoration completed" (or something like that) without an error, but the data does not get restored. In logcat and oandbackup.log there is no entry, but in oandbackup itself there is an error:

![screenshot_oandbackup_20180624-223416](https://user-images.githubusercontent.com/39731232/41823355-fd266570-77fe-11e8-8b12-b30d00d21e1f.png)

**Comment by simonsen23** [comment]
It is permissive.

**Comment by jensstein** [comment]
this apk has some more debugging (no fixes yet though). please send me a logcat.
https://gitlab.com/jensstein/oandbackup/-/jobs/80180769/artifacts/raw/apks/arm64/oandbackup-debug.apk

**Comment by simonsen23** [comment]
[oandlog.txt](https://github.com/jensstein/oandbackup/files/2213962/oandlog.txt)

Afaik the restoration from apk and data went well. When I have more time, I can try it with more apps.

I have one suggestion:
Everytime I install oandbackup I have to set the path for busybox manually (/system/xbin/busybox). Maybe you can add a check for this path too ;)

**Comment by jensstein** [comment]
yes it seems to have gone well. if you can confirm that it works with this apk i'll close the issue:
https://gitlab.com/jensstein/oandbackup/-/jobs/86828190/artifacts/file/oandbackup-signed-arm64.apk

i also added your suggestion about checking for /system/xbin/busybox :)

**Comment by simonsen23** [comment]
[log.txt](https://github.com/jensstein/oandbackup/files/2267932/log.txt)

With old backups it worked afaik well. But it won't create new backups.
I tried to look in the logfiles, but they seem empty (New issue?).

Unfortunately the check for the path didn't work :(
